### DISCUSSION TITLE
Enhancement: Nginx Unit endpoint /about

### DISCUSSION THREAD

**Discussion by Pavlusha311245** [nginx/unit#1092] [General]
Can an endpoint **/about** or other similarly named endpoint be added to display nginx unit version data? 
For example version, what socket or address is used and what configuration settings. 
Data can be obtained using the `unitd` and `unitd --version` commands.
I'd help with development, but I'm weak in C development 😔

**Comment by hongzhidao** [comment]
Hi @Pavlusha311245 
Thanks that you'd like to contribute on the project.

> Can an endpoint /about or other similarly named endpoint be added to display nginx unit version data?

You could refer to these first about the exported information. Make what information exported clear first.
a. exported ngx object in njs:
https://nginx.org/en/docs/njs/reference.html#ngx
b. exported ngx information in nginx plus.
https://nginx.org/en/docs/http/ngx_http_api_module.html#def_nginx_object

> Data can be obtained using the unitd and unitd --version commands.

I would suggest exporting it from unit status API. Adding an new object like "unit" at the top.
https://unit.nginx.org/usagestats/

> I'd help with development, but I'm weak in C development

You could have a try with it, I'd like to help you if possible.  Related source files:
```
src/nxt_status.h
src/nxt_status.c
build/include/nxt_version.h
build/include/nxt_auto_config.h
```

Btw, we need to discuss it internally about the feature's priority and if it's necessary to add into Unit.

**Comment by Pavlusha311245** [comment]
Can you @hongzhidao  put the issue label Enhancement?

> You could have a try with it, I'd like to help you if possible. Related source files:

I'll see what I can do.)

**Comment by Pavlusha311245** [comment]
@hongzhidao I'm in the process of doing this, but you can correct me if there is already a problem somewhere

**Comment by hongzhidao** [comment]
Hi @Pavlusha311245,
Now it's clear to provide `/status/unit` with information like `version`, others are not clear yet, but we'll discuss them internally.
```
curl http://127.1:8000/status
{
	"unit": {
		"version": "1.31.0"
	},

	...
}
```
I'm considering providing related configuration information like `load_time` and `generation`. For example:
```
{
	"unit": {
		"version": "1.31.0",
                "load_time": "2023-08-29T10:10:40.101Z",
                "generation": 12
	},

	...
}
```
`load_time` represents the last reconfiguration time.
`generation` is the total number of configuration reloads.

Would you like to have a try to support them?
Note the control process provides the status API, and the configuration also happens in the same process.

Thanks.

**Comment by Pavlusha311245** [comment]
Hi @hongzhidao
Of course, I'll try to add data as well. But it will take a lot of time for now. My main stack is PHP :)

**Comment by hongzhidao** [comment]
> 

No rush at all.
With the two information, it can help you understand how to handle time with a specific format, unit processes, especially controller process, and what happens about the reconfiguration. It might take you much time to understand them before implementation.

**Comment by Pavlusha311245** [comment]
@hongzhidao Hello after almost six months 😊. I was still able to find time and finish what I was doing. I understand that this is not ideal, so could you check my changes? `feature/unit_about_section`  branch in my forked repo

**Comment by hongzhidao** [comment]
Hi @Pavlusha311245 
Welcome back, would you mind creating a PR?
Btw, it would be helpful to describe the feature with the first comment of PR and add something like how the new metric looks like. Thanks.

**Comment by Pavlusha311245** [comment]
@hongzhidao morning. PR created https://github.com/nginx/unit/pull/1089
